admin/server/github.go

```diff
@@ -474,6 +464,98 @@ func (s *Server) githubWebhook(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+// githubRepoStatus is a http wrapper over [GetGithubRepoStatus]. It redirects to the grantAccessURL if there is no access.
+// It's implemented as a non-gRPC endpoint mounted directly on /github/post-auth-redirect.
+// TODO :: refactor this and [GetGithubRepoStatus] to call some common service function.
+func (s *Server) githubRepoStatus(w http.ResponseWriter, r *http.Request) {
```

Comment:
I'm also okay with calling the gRPC handler to avoid the duplication in this case. I see why it's not easy to move to `s.admin` – feels like knowledge of frontend URLs to redirect to should not reside in `s.admin`, but only in this package.